Snowflake.Data.Tests/IntegrationTests/SFConnectionIT.cs

```diff
@@ -1540,6 +1540,62 @@ public void TestInvalidProxySettingFromConnectionString()
             }
         }
 
+        [Test]
+        [TestCase("*")]
+        [TestCase("*{0}*")]
+        [TestCase("^*{0}*")]
+        [TestCase("*{0}*$")]
+        [TestCase("^*{0}*$")]
+        [TestCase("^nonmatch*{0}$|*")]
+        [TestCase("*a*", "a")]
+        [TestCase("*la*", "la")]
+        public void TestNonProxyHostShouldBypassProxyServer(string regexHost, string proxyHost = "proxyserverhost")
+        {
+            using (var conn = new SnowflakeDbConnection())
+            {
+                // Arrange
+                var host = ResolveHostToUseFromTestConfig(ConnectionString);
+                var nonProxyHosts = string.Format(regexHost, $"{host}");
+                conn.ConnectionString =
+                    $"{ConnectionString}USEPROXY=true;PROXYHOST={proxyHost};NONPROXYHOSTS={nonProxyHosts};PROXYPORT=3128;";
+                
+                // Act
+                conn.Open();
+                
+                // Assert
+                // If the connection is open, it indicates that the server proxy was bypassed.
+                Assert.AreEqual(ConnectionState.Open, conn.State);
+            }
+        }
+
+        [Test]
+        [TestCase("invalid{0}")]
+        [TestCase("*invalid{0}*")]
+        [TestCase("^invalid{0}$")]
+        [TestCase("*a.b")]
+        [TestCase("a", "a")]
+        [TestCase("la", "la")]
+        public void TestNonProxyHostShouldNotBypassProxyServer(string regexHost, string proxyHost = "proxyserverhost")
+        {
+            using (var conn = new SnowflakeDbConnection())
+            {
+                // Arrange
+                var nonProxyHosts = string.Format(regexHost, $"{testConfig.host}");
+                conn.ConnectionString =
+                    $"{ConnectionString}connection_timeout=5;USEPROXY=true;PROXYHOST={proxyHost};NONPROXYHOSTS={nonProxyHosts};PROXYPORT=3128;";
+                
+                // Act/Assert
+                // If the nonproxyhosts property fails to bypass the server, the proxy will be use but it will return a exception
+                // because is not configure in the testing environment so the exception is used to verify that was not bypass.
+                var exception = Assert.Throws<SnowflakeDbException>(() => conn.Open());
+                
+                // Assert
+                s_logger.Debug("Failed opening connection ", exception);
```

Comment:
it is OK, but normally we don't add logging to tests because it doesn't facilitate anything.
This log will not be visible anywhere because we have WARN log level set by default.
If we would like to check what the exception was exactly we could add a breakpoint here, run the test in debug mode and see the exception.
So I personally would not add such a debug statements, but I see that it was used in this class a few times so it could stay, but personally I don't like such things.

Reply:
Removed logger, thank you!!